dm/worker/worker.go

```diff
@@ -80,8 +80,14 @@ func NewWorker(cfg *Config) (*Worker, error) {
 	}
 	w.relayPurger = purger
 
-	// open kv db
+	// try upgrade from an older version
 	dbDir := path.Join(w.cfg.MetaDir, "kv")
+	err = tryUpgrade(dbDir)
+	if err != nil {
+		return nil, errors.Annotatef(err, "try to upgrade from any older version to %v", newCurrentVersion())
```

Comment:
error message or return values can contain upgrade from which version to which, or do we have a log about elsewhere

Reply:
older version information (which one/whether exist) is included in `err`.